cedar-policy-validator/src/diagnostics/validation_errors.rs

```diff
@@ -460,28 +460,19 @@ impl Diagnostic for FunctionArgumentValidation {
 
 /// Structure containing details about a hierarchy not respected error
 #[derive(Debug, Clone, Hash, Eq, PartialEq, Error)]
-#[error("for policy `{policy_id}`, operands to `in` do not respect the entity hierarchy")]
+#[error("Internal invariant violated: `HierarchyNotRespected` error should never occur. Please file an issue")]
 pub struct HierarchyNotRespected {
     /// Source location
```

Comment:
`wrap_core_error!` in `api/err/validation_errors.rs` needs them.  Taking a couple steps back, since the variant still exists in the public `ValidationError`, we need to support `ValidationError::policy_id()` on it.  And you could say we could return a fictitious policy id or something, but if this error actually ever does get thrown in violation of the internal invariant (due to some future refactor), we'll be glad to have the policy id and source location